sql-statements/sql-statement-select.md

```diff
@@ -31,7 +31,23 @@ SelectStmtOpts ::=
     SelectStmtStraightJoin
 
 TableRefsClause ::=
-    TableRef AsOfClause? ( ',' TableRef AsOfClause? )*
+    TableRef ( ',' TableRef )*
+
+TableRef ::=
+    TableFactor
+|   JoinTable
+
+TableFactor ::=
+    TableName ( "PARTITION" "(" PartitionName ")" )? ("AS" TableAlias)? AsOfClause? TableSample?
+
+JoinTable ::=
+    ( TableRef ( ("INNER" | "CROSS")? "JOIN" | "STRAIGHT_JOIN" ) TableFactor JoinClause?
+    | TableRef ( ("LEFT" | "RIGHT") "OUTER"? "JOIN" ) TableRef JoinClause
+    | TableRef "NATURAL" "INNER"? ("LEFT" | "RIGHT") "OUTER"? "JOIN" ) TableFactor
```

Comment:
TiDB's parser.y is different from MySQL's grammar. 

1. it support the following combinations:

| join type \ join clause | empty | ON expr | USING (columns) |
|---|----|----|----|
| ('CROSS' &#x7c; 'INNER')? 'JOIN' | ✓ |  ✓ |  ✓ | 
| ('LEFT' &#x7c; 'RIGHT') 'OUTER'? 'JOIN' | ✗ | ✓ |  ✓ | 
| 'NATURAL' (('LEFT' &#x7c; 'RIGHT') 'OUTER'?) 'JOIN' | ✓ | ✗ | ✗ |
| 'STRAIGHT_JOIN' | ✓ | ✓ | ❌ |

In particular `t1 STRAIGHT_JOIN t2 USING (col)` is syntax error, incompatible with MySQL. This is probably a bug.

2. The RHS of all join operations are all `TableRef`, not `TableFactor`.
3. TiDB does not support the syntax `NATURAL INNER JOIN`, again incompatible with MySQL. Your current production also permits the nonsense `NATURAL INNER OUTER JOIN`. If we follow MySQL's rule it should has been `'NATURAL' ('INNER' | ('LEFT' | 'RIGHT') 'OUTER'?)? 'JOIN'`.